oat-sa/extension-tao-testqti
# Integration/rfe 1247 m3/solar design feature hiding

related to 
 - https://oat-sa.atlassian.net/browse/AUT-3623
 - https://oat-sa.atlassian.net/browse/AUT-3625

⚠️ Update extension version to the release ones before merging

###Description

This PR addresses requirement 3.h from the ticket to hide various test props if solar design is activated.

This is an integration branch from FE and BE changes :

FE: https://github.com/oat-sa/extension-tao-testqti/pull/2475
BE: https://github.com/oat-sa/extension-tao-testqti/pull/2481


### How to test

To test locally the dependent FE/BE branches should be specified in `composer.json`
 - [feat/AUT-3625/content-creation-mode-switch](https://github.com/oat-sa/generis/tree/feat/AUT-3625/content-creation-mode-switch)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/tao-core/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/extension-tao-deliver-connect/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [integration/RFE-1247-M3/solar-design-feature-hiding](https://github.com/oat-sa/extension-tao-testqti/tree/integration/RFE-1247-M3/solar-design-feature-hiding)
 - [feat/AUT-3623/hide-features-in-solar-design](https://github.com/oat-sa/extension-tao-delivery-rdf/tree/feat/AUT-3623/hide-features-in-solar-design)
 - [feat/AUT-3623/hide-features-in-solar-design](https://github.com/oat-sa/extension-tao-itemqti/tree/feat/AUT-3623/hide-features-in-solar-design)

Deployed to kitchen env, so functionally can be validated there: http://test-rfe-1147-facelift.playground.kitchen.it.taocloud.org:49381/

List of the related PRs (this PR doesn't depend on them, but can be reviewed together):

 - https://github.com/oat-sa/extension-tao-delivery-rdf/pull/484
 - https://github.com/oat-sa/extension-tao-deliver-connect/pull/204
 - https://github.com/oat-sa/extension-tao-itemqti/pull/2511
 - https://github.com/oat-sa/tao-core/pull/4034


### How to test

ℹ️ Backend part is not available yet. To activate feature hiding you need to update the configuration in `tao/config/tao/client_lib_config_registry.conf.php`, then feature hiding can be validated

```
        'services/features' => array(
            'visibility' => array(
                'taoQtiItem/creator/item/property/identifier' => 'hide',
                'taoQtiItem/creator/interaction/response/property/identifier' => 'hide',
                'taoQtiItem/creator/interaction/response/responseProcessing/custom' => 'hide',
                'taoQtiItem/creator/interaction/choice/property/orientation' => 'hide',
                'taoQtiItem/creator/interaction/extendedText/property/constraints' => 'hide',
                'taoQtiItem/creator/static/text/scrolling' => 'hide',
                'taoQtiItem/creator/interaction/textEntry/property/base' => 'hide',
                'taoQtiItem/creator/interaction/textEntry/property/constraints' => 'hide',
                'taoQtiItem/creator/interaction/textEntry/property/recommendations' => 'hide',
                'taoQtiItem/creator/interaction/pci/liquidsInteraction' => 'hide',
                'taoDeliverConnect/publish/singleTenant' => 'hide',
                'taoQtiTest/creator/test/property/identifier' => 'hide',
                'taoQtiTest/creator/test/property/lateSubmission' => 'hide',
                'taoQtiTest/creator/test/property/scoring/custom' => 'hide',
                'taoQtiTest/creator/test/property/scoring/weight' => 'hide',
                'taoQtiTest/creator/test/property/outcomeDeclarations' => 'hide',
                'taoQtiTest/creator/testPart/property/identifier' => 'hide',
                'taoQtiTest/creator/testPart/property/lateSubmission' => 'hide',
                'taoQtiTest/creator/testPart/property/itemSessionControl' => 'hide',
                'taoQtiTest/creator/testPart/category/presetGroup/warning' => 'hide',
                'taoQtiTest/creator/section/property/identifier' => 'hide',
                'taoQtiTest/creator/section/property/lateSubmission' => 'hide',
                'taoQtiTest/creator/section/property/visible' => 'hide',
                'taoQtiTest/creator/section/property/keepTogether' => 'hide',
                'taoQtiTest/creator/section/property/itemSessionControl/validateResponses' => 'hide',
                'taoQtiTest/creator/section/property/itemSessionControl/showFeedback' => 'hide',
                'taoQtiTest/creator/section/property/rubricBlocks/class' => 'hide',
                'taoQtiTest/creator/itemRef/property/identifier' => 'hide',
                'taoQtiTest/creator/itemRef/property/reference' => 'hide',
                'taoQtiTest/creator/itemRef/property/lateSubmission' => 'hide',
                'taoQtiTest/creator/itemRef/category/presetGroup/navigation' => 'hide',
                'taoQtiTest/creator/itemRef/property/itemSessionControl/showFeedback' => 'hide',
                'taoQtiTest/creator/itemRef/property/itemSessionControl/validateResponses' => 'hide',
                'taoQtiTest/creator/itemRef/property/weights' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/resourceIdentifier' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/maxExecutions' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/startDate' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/endDate' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/displayOrder' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/access' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/proctoringSettings' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/publicationId' => 'hide',
                'taoDeliveryRdf/deliveryMgmt/assessmentProjectId' => 'hide'
            )
        ),
```
